javaagent-core/build.gradle.kts

```diff
@@ -9,5 +9,13 @@ dependencies {
     api("io.opentelemetry:opentelemetry-api:${versions["opentelemetry"]}")
     api("io.opentelemetry.javaagent:opentelemetry-javaagent-instrumentation-api:${versions["opentelemetry_java_agent"]}")
     implementation("org.slf4j:slf4j-api:${versions["slf4j"]}")
-    implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.13.3")
+    implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.13.3") {
+        constraints {
```

Comment:
nice :)

Reply:
For my knowledge, what does the constraint mean and do? @bradAtTraceable

Reply:
My understanding is that the constraint forces Gradle to use a particular version of a transitive dependency.  So, in this case, the artifact that we are building has a direct dependency on "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml", specifically the 2.13.3 version.  The dataformat-yaml component has a dependency on "org.yaml:snakeyaml".  Apparently, snyk has detected that there was an exposure with snakeyaml, which was fixed in version 1.31.  The constraint forces gradle to use the 1.31 version of snakeyaml, so that snyk will no longer block the build from completing.

Reply:
@ryanericson  I should mention that this is the preferred method of dealing with snyk issues, as opposed to merely forcing snyk to ignore the issue.